System/Win32/MinTTY.hsc

```diff
@@ -0,0 +1,207 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+#if __GLASGOW_HASKELL__ >= 709
+{-# LANGUAGE Safe #-}
+#elif __GLASGOW_HASKELL__ >= 701
+{-# LANGUAGE Trustworthy #-}
+#endif
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  System.Win32.MinTTY
+-- Copyright   :  (c) University of Glasgow 2006
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  Esa Ilari Vuokko <ei@vuokko.info>
+-- Stability   :  provisional
+-- Portability :  portable
+--
+-- A function to check if the current terminal uses MinTTY.
+-- Much of this code was originally authored by Phil Ruffwind.
+--
+-----------------------------------------------------------------------------
+
+module System.Win32.MinTTY (isMinTTY) where
+
+import Graphics.Win32.Misc
+import System.Win32.DLL
+import System.Win32.File
+import System.Win32.Types
+
+#if MIN_VERSION_base(4,6,0)
+import Control.Exception (catch)
+#endif
+import Control.Monad (void)
+import Data.List (isInfixOf, isSuffixOf)
+import Foreign hiding (void)
+import Foreign.C.String
+import Foreign.C.Types
+
+#if __GLASGOW_HASKELL__ < 711
+#let alignment t = "%lu", (unsigned long)offsetof(struct {char x__; t (y__); }, y__)
+#endif
+
+##include "windows_cconv.h"
+#include <windows.h>
+
+-- | Returns 'True' is running on a MinTTY console (e.g., Cygwin or MSYS).
+-- Returns 'False' otherwise.
+isMinTTY :: IO Bool
+isMinTTY = do
+    h <- getStdHandle sTD_ERROR_HANDLE
```

Comment:
This currently just hardwires in `sTD_ERROR_HANDLE`. Do you think that's acceptable, or should we provide a version of `isMinTTY` that accepts any `HANDLE`?